ci/pipeline

```diff
@@ -273,9 +285,79 @@ def release(requestVersion: String, gitSha: String, latest: Boolean = false): Un
   // TODO: Publish native packages to unstable.
 }
 
+/**
+  * Given tagName v1.5.1, and rev DEADBEEF, we would assert that v1.5.0 is a direct ancestor of rev DEADBEEF
+  *
+  * Also, we assert that rev contains new commits since the last release. If the proposal is to tag v1.5.1 with same
+  * revision as v1.5.0, then we fail, saying that there v1.5.0 points to the commit porposed for v1.5.1
+  *
+  * @param tagName The tag name / version we are going to tag. eg v1.5.1
+  * @param gitSha The commit sha / rev that will be tagged.
+  */
+def assertPriorVersionIsAncestor(tagName: String, gitSha: String): Unit = {
```

Comment:
Why is this required?

Reply:
this ensures that the sha that is being released is on the correct release branch.   It is designed to prevent human error... because I understand that humans like errors so much they create them or something

Reply:
Ah, so I cannot release a 1.6 commit as 1.5? Could we express this in the comments?

Reply:
it seems like it is there but I made a clarification statement.